text/0000-template-comments.md

```diff
@@ -0,0 +1,86 @@
+---
+title: HTML comments for LWC templates
+status: DRAFTED
+created_at: 2021-03-31
+updated_at: 2021-03-31
+pr: https://github.com/salesforce/lwc/pull/2263
+---
+
+# HTML comments for LWC templates
+
+## Summary
+
+In most cases, HTML comments in templates only add information for other developers. There are some use cases where HTML comments convey some runtime semantic (e.g., MSO comments). As of today, the LWC template compiler strips HTML comments. This RFC proposes a way to preserve HTML comments in LWC templates.
+
+### Basic example
+
+```html
+<template preserve-comments>
+  <!-- Greeter container -->
+  <div>
+	Hello <b>{name}</b>!
+  </div>
+  <!-- eof: Greeter container -->
+</template>
+```
+```html
+<x-example>
+  # shadow-root
+  |   <!-- Greeter container -->
+  |   <div>
+  |     Hello <b>world</b>
+  |   </div>
+  |   <!-- eof: Greeter container -->
+</x-example>
+```
+
+## Motivation
+
+Microsoft Outlook's desktop email client leverage MSO (Microsoft Office) to render their HTML content. Because of this, the support for common HTML and CSS properties is limited, eg: `<div>`s are not supported.
+
+Email developers rely on [MSO conditionals](https://docs.microsoft.com/en-us/previous-versions/windows/internet-explorer/ie-developer/compatibility/ms537512(v=vs.85)?redirectedfrom=MSDN)  to improve the UX of emails opened on Microsoft Outlook.
+
+This RFC introduces a new opt-in mechanism at compile time to preserve comments on the rendered component.
+
+## Proposal
+
+### Compile-time behavior
+
+LWC compilation strips HTML comments because the browsers do not use them at runtime. The bundle's size is smaller by stripping the HTML comments, and the rehydration process does not need to take into account comments nodes.
+
+To preserve the performance of the existing components, we propose to introduce one of these:
```

Comment:
What about both options?

The main advantage of the attribute is that it allows platform developers to benefit from this feature since they don't have access to the LWC compiler config. When you have access to the compiler config (eg. OSS usage) a single flag is also convinient.

Reply:
I like it, adding it.